src/test/java/tests/BaseTest.java

```diff
@@ -0,0 +1,42 @@
+package tests;
+
+import org.openqa.selenium.WebDriver;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import steps.NavigationSteps;
+import steps.SearchSteps;
+import utils.DriverFactory;
+import utils.PropertyReader;
+
+import java.time.Duration;
+
+public abstract class BaseTest {
+
+    private static WebDriver driver;
+    NavigationSteps navigationStep;
+    SearchSteps searchSteps;
+
+    public static WebDriver getDriver() {
```

Comment:
please create a DriverManager class to manage all the actions with webDriver.
gettingDriver in BaseTest is not the best approach because for some purposes you might ned to use it inside PageObjects or Steps but with current architecture implementation it would be wrong

Reply:
fixed